backend/tests/unit/trigger/test_catalogue.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pytest
+
+from infrahub.trigger import catalogue
+from infrahub.trigger.catalogue import builtin_triggers
+
+if TYPE_CHECKING:
+    from infrahub.trigger.models import TriggerDefinition
+
+
+@pytest.mark.parametrize("trigger", [pytest.param(trigger, id=trigger.name) for trigger in builtin_triggers])
+def test_builtin_trigger_definition(trigger: TriggerDefinition) -> None:
+    """Validate that the actions associated with the trigger matches the definition of the workflow."""
+    trigger.validate_actions()
```

Comment:
This will ensure that the triggers are only using parameters supported by the associated workflow